aspnet/mvc/views/layout.rst

```diff
@@ -0,0 +1,98 @@
+
+Layouts can be nested; each layout can itself specify another layout. When rendering a view, :ref:`ViewStart <viewstart>` is run first, followed by the view itself, followed by its layout(s).
+
+Every layout must call ``RenderBody`` . Wherever the call to ``RenderBody`` is placed, the contents of the view will be rendered.
+
+Sections
+^^^^^^^^
+
+A layout can optionally reference one or more *sections*, by calling ``RenderSection``. Each call to ``RenderSection`` can specify whether that section is required or optional. Individual views specify the content to be rendered within a section using the ``@section`` razor syntax. If a view defines a section, it must be rendered (or an error will occur).
+
+An example ``@section`` definition in a view:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/Manage/VerifyPhoneNumber.cshtml
+  :language: html
+  :lines: 28-30
+
+Sections only flow from views. They cannot be referenced from partials, view components, or other parts of the view system.
+
+.. _viewimports:
+
+Importing Shared Directives
+---------------------------
+
+Views can use razor directives to specify namespaces or perform :doc:`dependency injection <dependency-injection>`. Directives used by many views may be specified in a ``_ViewImports.cshtml`` file. The ``_ViewImports`` file supports the following directives:
+
+- addTagHelper
+- removeTagHelper
+- tagHelperPrefix
+- using
+- model
+- inherits
+- inject
+
+The file does not support other razor features, such as functions and section definitions.
+
+A sample ``_ViewImports`` file:
+
+.. literalinclude:: ../../../common/samples/WebApplication1/src/WebApplication1/Views/_ViewImports.cshtml
+  :language: html
+
+The ``_ViewImports`` file for an ASP.NET Core MVC app is typically placed in the ``Views`` folder root. A ``_ViewImports`` file can be placed within a controller-associated view folder, in which case it will only be applied to views within that folder. ``_ViewImports`` files are run first at the root level, and then for a ``_ViewImports`` file specified in the controller-associated folder, so settings specified at the root level may be overridden at the folder level.
+
+.. _viewstart:
+
+Running Code Before Each View
+-----------------------------
+
+If you have code you need to run before every view, this should be placed in the ``_ViewStart.cshtml`` file. By convention, the ``_ViewStart.cshtml`` file is located in the root of the ``Views`` folder. The statements listed in ``ViewStart`` are run before every full view (not layouts, and not partial views). Like :ref:`ViewImports <viewimports>`, ``_ViewStart`` is hierarchical. If a ``_ViewStart`` file is defined in the controller-associated view folder, it will be run after the of one defined in the root of the ``Views`` folder (if any).
```

Comment:
- Be consistent with your `_ViewStart.cshtml` / `_ViewStart` / `ViewStart` usage. Could do `_ViewStart` or `_ViewStart.cshtml`